### PR TITLE
fix(types): fixes `sorter` param types

### LINKED ISSUE #36653: ts error  in Table  onChange's sorter parameter
### Reproduction link

[https://codesandbox.io/s/antd-reproduction-template-forked-mgi7el?file=/index.tsx](https://codesandbox.io/s/antd-reproduction-template-forked-mgi7el?file=/index.tsx)

### Steps to reproduce

When:
```
  onChange={(pagination, filter, sorter) => {
            setQuery({
              ...query,
              sortField: sorter.field,
              sortOrder: sorter.order,
```

### What is expected?

ts error passed , when I defined Column's type

### What is actually happening?

Property 'field' does not exist on type 'SorterResult<T> | SorterResult<T>[]'.
  Property 'field' does not exist on type 'SorterResult<T>[]'.ts(2339)

| Environment | Info |
| --- | --- |
| antd | 4.21.6 |
| React | 17.0.2 |
| System | MacOS |
| Browser | Chrome: Version 103.0.5060.114 (Official Build) (arm64) |

### PR DESCRIPTION
**Pull request by kungege**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #36653

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Table `onChange` argument `sorter` type error. |
| 🇨🇳 Chinese | 修复 Table `onChange` 参数 `sorter` 类型报错的问题。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
Could you add a type test case in `components/table/__tests__`.

**Comment by kungege** [comment]
> Could you add a type test case in `components/table/__tests__`?

Sorry about that. I don‘t know how to add a `type` test case.

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/blob/master/components/table/__tests__/type.test.tsx

https://github.com/ant-design/ant-design/pull/32358/files#diff-c7bb8ec59175366c52da6ca55ff47144b4328563bc2742cc18809362a8d40b4fR32

**Comment by kungege** [comment]
> See https://github.com/ant-design/ant-design/blob/master/components/table/__tests__/type.test.tsx
> 
> https://github.com/ant-design/ant-design/pull/32358/files#diff-c7bb8ec59175366c52da6ca55ff47144b4328563bc2742cc18809362a8d40b4fR32
Thank you! I will have a try.

File: components/table/__tests__/type.test.tsx [hunk 1]
```diff
@@ -36,6 +36,35 @@ describe('Table.typescript', () => {
...
+            value: 'Jim',
+          },
+        ],
+        sorter: (a, b) => a.name.length - b.name.length,
+      },
+    ];
+
+    const table = <Table columns={columns} onChange={onChange} />;
+
+    expect(table).toBeTruthy();
```
Comment (afc163) [inline components/table/__tests__/type.test.tsx:66]: 这个例子重现不了问题。期望是删掉 components/table/hooks/useSorter.tsx 的改动后，这里会报错。

Comment (afc163) [reply components/table/__tests__/type.test.tsx:66]: ping

**Comment by MadCcc** [comment]
https://github.com/ant-design/ant-design/issues/37099
https://github.com/ant-design/ant-design/issues/37024

Break

**Pull request merged**